pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -88,26 +92,49 @@ func (l lazyLoadedHeadersSnapshot) persist(persistDir string) error {
 
 // NewReaderPool makes a new ReaderPool. If lazy-loading is enabled, NewReaderPool also starts a background task for unloading idle Readers and persisting a list of loaded Readers to disk.
 func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTimeout time.Duration, metrics *ReaderPoolMetrics, lazyLoadedSnapshotConfig LazyLoadedHeadersSnapshotConfig) *ReaderPool {
-	p := newReaderPool(logger, lazyReaderEnabled, lazyReaderIdleTimeout, metrics)
+	var snapshot *lazyLoadedHeadersSnapshot
+	if lazyReaderEnabled && lazyLoadedSnapshotConfig.EagerLoadingEnabled {
+		lazyLoadedSnapshotFileName := filepath.Join(lazyLoadedSnapshotConfig.Path, lazyLoadedHeadersListFile)
+		var loadSnapshotErr error
+		snapshot, loadSnapshotErr = loadLazyLoadedHeadersSnapshot(lazyLoadedSnapshotFileName)
+		if loadSnapshotErr != nil {
+			level.Warn(logger).Log("msg", "loading the list of index-headers from file failed; starting with no loaded index-headers", "file", lazyLoadedSnapshotFileName, "err", err)
```

Comment:
Shouldn't we log the error from loading the snapshot? (Should we rename `loadSnapshotErr` to `err`? Not sure it needs to be special.)

Also, given snapshots are per-tenant, should we clarify that we're only skipping loading index-headers for a particular tenant? 

```suggestion
			level.Warn(logger).Log("msg", "loading the list of index-headers from file failed; not eagerly loading index-headers for tenant", "file", lazyLoadedSnapshotFileName, "err", loadSnapshotErr, "tenant", XXX)
```

Reply:
There were some refactor renaming issues. I fixed it, just use `err` for the error name and applied the updated log message suggestion.